crates/driver/src/boundary/liquidity/uniswap/v3.rs

```diff
@@ -0,0 +1,132 @@
+        liquidity_collector::LiquidityCollecting,
+    },
+    std::sync::{Arc, Mutex},
+};
+
+pub fn to_domain(id: liquidity::Id, pool: ConcentratedLiquidity) -> liquidity::Liquidity {
+    let handler = pool
+        .settlement_handling
+        .as_any()
+        .downcast_ref::<uniswap_v3::UniswapV3SettlementHandler>()
+        .expect("downcast uniswap settlement handler");
+
+    liquidity::Liquidity {
+        id,
+        gas: eth::Gas(pool.pool.gas_stats.mean_gas),
+        kind: liquidity::Kind::UniswapV3(Pool {
+            router: handler.inner.router.address().into(),
+            address: pool.pool.address.into(),
+            tokens: liquidity::TokenPair::new(
+                pool.pool.tokens[0].id.into(),
+                pool.pool.tokens[1].id.into(),
+            )
+            .unwrap(),
+            sqrt_price: SqrtPrice(pool.pool.state.sqrt_price),
+            liquidity: Liquidity(pool.pool.state.liquidity.as_u128()),
+            tick: Tick(pool.pool.state.tick.to_i32().unwrap()),
+            liquidity_net: pool
+                .pool
+                .state
+                .liquidity_net
+                .iter()
+                .map(|(key, value)| {
+                    (
+                        Tick(key.to_i32().unwrap()),
+                        LiquidityNet(value.to_i128().unwrap()),
+                    )
+                })
+                .collect(),
+            fee: Fee(pool.pool.state.fee),
+        }),
+    }
+}
+
+pub fn to_interaction(
+    pool: &liquidity::uniswap::v3::Pool,
+    input: &liquidity::MaxInput,
+    output: &liquidity::ExactOutput,
+    receiver: &eth::Address,
+) -> eth::Interaction {
+    let web3 = shared::ethrpc::dummy::web3();
+
+    let handler = UniswapV3SettlementHandler::new(
+        UniswapV3SwapRouter::at(&web3, pool.router.0),
+        GPv2Settlement::at(&web3, receiver.0),
+        Mutex::new(Allowances::empty(receiver.0)),
+        pool.fee.0,
+    );
+
+    let (_, interaction) = handler.settle(
+        TokenAmount::new(input.0.token.into(), input.0.amount),
+        TokenAmount::new(output.0.token.into(), output.0.amount),
+    );
+
+    interaction
+        .encode()
+        .into_iter()
+        .map(|(target, value, call_data)| eth::Interaction {
+            target: eth::Address(target),
+            value: eth::Ether(value),
+            call_data: call_data.0,
+        })
+        .next()
+        .expect("returns exactly 1 interaction")
+}
+
+pub async fn collector(
+    eth: &Ethereum,
+    block_retriever: Arc<dyn BlockRetrieving>,
+    config: &infra::liquidity::config::UniswapV3,
+) -> Result<Box<dyn LiquidityCollecting>> {
+    let web3 = boundary::web3(eth);
+    let router = UniswapV3SwapRouter::at(&web3, config.router.0);
+
+    let pool_fetcher = Arc::new(
+        UniswapV3PoolFetcher::new(
+            eth.chain_id().0.as_u64(),
+            web3.clone(),
+            // TODO: pass client into this function?
+            Default::default(),
```

Comment:
Not sure I understand this TODO.

Reply:
It's unclear to me if we need to pass a special `reqwest::Client` into the function or if it is okay to just create it in place. This would affect things like timeout limits for example.

Reply:
Ah OK, makes sense. Can you add a bit more context to the comment (I think what you wrote here in the PR is enough).

Reply:
added: [c4d033a](https://github.com/cowprotocol/services/pull/1251/commits/c4d033aa665a6a73539aa352488ec03d11720bb5)